modelName not modelType

y-tang opened the pull request:
@shelbys review please

OwenBrotherwood committed:
access-token: add option "searchDefaultTokenKeys"

Set this option to false to prevent AccessToken from checking default
places like "access_token" in query.

Commit:
Add a unit-test for searchDefaultTokenKeys

Commit:
fix exit early when password is non-string
closes #1437

PradnyaBaviskar committed:
Promisify User model

digitalsadhu committed:
Add missing . to user model property descriptions

digitalsadhu committed:
Create stack-removing errorhandler middleware

Create a wrapper around express' errorhandler that removes the stack
trace from error responses when "options.includeStack" is false.

Commit:
Update model.js

Updated JSdoc for remoteMethod

ritch committed:
Remove trailing whitespace from jsdoc

bajtos committed:
Merge pull request #1416 from strongloop/feature/searchDefaultTokenKeys

Config option: (do not) search default token keys

ritch committed:
Add PersistedModel.createChangeStream()

ritch committed:
Merge pull request #1445 from strongloop/feature/subscribe

Add PersistedModel.subscribe()

ritch committed:
2.19.0

 * Add PersistedModel.createChangeStream() (Ritchie Martori)

 * Remove trailing whitespace from jsdoc (Ritchie Martori)

 * Update model.js (Rand McKinney)

 * Downgrade version of loopback-testing (Ritchie Martori)

 * Auto-configure models required by `app.enableAuth` (Miroslav Bajtoš)

 * Add loadBuiltinModels flag to loopback(options) (Miroslav Bajtoš)

 * Add a unit-test for searchDefaultTokenKeys (Miroslav Bajtoš)

 * access-token: add option "searchDefaultTokenKeys" (Owen Brotherwood)

 * Fix the test case (Raymond Feng)

 * Fix code standards issues (Tom Kirkpatrick)

 * Add test case to highlight fatal error when trying to include a scoped relationship through a polymorphic relationship (Tom Kirkpatrick)

 * add callback args for listByPrincipalType to jsdoc comment, pass explicit arguments to callback (Esco Obong)

 * mark utiltiy function as private (Esco Obong)

 * fix linting errors (Esco Obong)

 * fix lint erros (Esco Obong)

 * consolidate Role methods roles, applications, and users into one, add query param to allow for pagination and restricting fields (Esco Obong)

 * fix implementation of Role methods: users,roles, and applications (Esco Obong)

ritch committed:
Add link to createChangeStream docs

PradnyaBaviskar committed:
Promisify 'Application' model

PradnyaBaviskar committed:
Promisify 'PersistedModel - replication'

Commit:
Fix jsdocs for methods with where argument

Commit:
Disable application model test for karma

Commit:
2.19.1

 * Disable application model test for karma (Raymond Feng)

 * Fix jsdocs for methods with where argument (Raymond Feng)

 * Add link to createChangeStream docs (Ritchie Martori)

raymondfeng committed:
Merge pull request #1465 from FreeCodeCamp/fix/no-password

fix: exit early when password is non-string

bajtos committed:
Merge pull request #1493 from PradnyaBaviskar/issue418-userModel

Promisify User model

Commit:
Don't load Bluebird for createPromiseCallback

The decision which Promise implementation to use should be made by
LoopBack user, not by the framework.

This commit moves Bluebird reference from lib/utils.js to
test/support.js.

bajtos committed:
Merge pull request #1560 from strongloop/fix/forgotten-bluebird-reference

Don't load Bluebird for createPromiseCallback

Commit:
Allow methods filter for middleware config

bajtos committed:
Merge pull request #1501 from digitalsadhu/master

Add missing . to user model property descriptions

raymondfeng committed:
Merge pull request #1561 from strongloop/feature/allow-middleware-methods

Allow methods filter for middleware config

Commit:
2.20.0

 * Allow methods filter for middleware config (Raymond Feng)

 * Don't load Bluebird for createPromiseCallback (Miroslav Bajtoš)

 * fix exit early when password is non-string closes #1437 (Berkeley Martinez)

 * Promisify User model (Pradnya Baviskar)

 * Add missing . to user model property descriptions (Richard Walker)

Commit:
Merge pull request #1533 from PradnyaBaviskar:issue418-application

Promisify 'Application' model

Close #1533

Commit:
Merge pull request #1536 from PradnyaBaviskar:issue418-replication

Promisify 'PersistedModel - replication'

Close #1536

Commit:
Add util methods to ACL and clean up related model resolutions

raymondfeng committed:
Merge pull request #1584 from strongloop/feature/add-more-acl-utils

Enhance the ACL related models

Commit:
2.21.0

 * Add util methods to ACL and clean up related model resolutions (Raymond Feng)

 * Promisify 'PersistedModel - replication' (Pradnya Baviskar)

 * Promisify 'Application' model (Pradnya Baviskar)

Commit:
fix typo "PeristedModel"

Commit:
Upgrade Travis to container-based infrastructure

See http://docs.travis-ci.com/user/migrating-from-legacy/ for details.

bajtos committed:
Merge pull request #1618 from strongloop/fix/travis-config

Upgrade Travis to container-based infrastructure

faridnsh committed:
Fix options.to assertion message in user.verify

superkhau committed:
Merge pull request #1624 from alFReD-NSH/patch-1

Fix options.to assertion message in user.verify

gausie committed:
Allow EJS templates to use includes

Simple change that allows templates to use includes.

bajtos committed:
Merge pull request #1593 from webvariants/master

fix typo "PeristedModel"

bajtos committed:
Merge pull request #1632 from gausie/patch-2

Allow EJS templates to use includes

Commit:
Update README.md

Fix link to loopback.io so it shows as link on npmjs.

Commit:
Merge pull request #1502 from digitalsadhu/suppress_stack_traces_in_production#564

Suppress error handler stack traces in production

Close #1502
Fix #564

Commit:
2.22.0

 * Create stack-removing errorhandler middleware (Richard Walker)
 * Update README.md (Rand McKinney)
 * Allow EJS templates to use includes (Samuel Gaus)
 * Fix options.to assertion message in user.verify (Farid Nouri Neshat)
 * Upgrade Travis to container-based infrastructure (Miroslav Bajtoš)
 * fix typo "PeristedModel" (Christoph)

superkhau committed:
Stop NPM license warning

superkhau committed:
Merge pull request #1656 from strongloop/feature/stop-license-warning

Stop NPM license warning

Commit:
Update persisted-model.js

Fix links in API docs to docs for where filter

Commit:
Update persisted-model.js

Remove trailing whitespace.

felipecrv committed:
Update comment about user ACL to reflect implementation

Commit:
Fix perf of rectification after updateAttributes

Improve the id-detection algorithm in the "after save" hook
to correctly handle "updateAttributes" as a single-model change
and DO NOT trigger full "rectify all" scan.

gausie committed:
Do not include redundant ports in verify links

If the protocol and port match we can ignore the port for a more
visually appealing link.

bajtos committed:
Merge pull request #1681 from strongloop/fix/replication-perf-on-save

Fix perf of rectification after updateAttributes

Commit:
2.22.1

 * Fix perf of rectification after updateAttributes (Miroslav Bajtoš)
 * Update persisted-model.js (Rand McKinney)
 * Stop NPM license warning (Simon Ho)

superkhau committed:
Add test fixtures for shared methods

superkhau committed:
Check configs for shared method settings

sam-github committed:
Set package license to MIT

sam-github committed:
Use strongloop conventions for licensing

Commit:
2.22.2

 * Use strongloop conventions for licensing (Sam Roberts)
 * Set package license to MIT (Sam Roberts)

superkhau committed:
Clean up .jshintrc

superkhau committed:
Merge pull request #1699 from strongloop/feature/clean-up-jshintrc

Clean up .jshintrc

superkhau committed:
Merge pull request #1667 from strongloop/bug/1322

Return HTTP 200 when deleting a resource via REST

wusuopu committed:
Set application's id property only if it's empty.

Fix `Application.resetKeys()` to reset instance id only if it is not
already set. This fixes a bug where each call of resetKeys created
a new instance.

bajtos committed:
Merge pull request #1668 from philix/comment

Update comment about user ACL to reflect implementation

Commit:
Merge pull request #1716 from wusuopu/feature/fix-issue-1689

Set application's id property only if it's empty.

Close #1716
Fix #1689

Commit:
Merge pull request #1684 from gausie/patch-3

Do not include redundant ports in verify links

Close #1684

Commit:
Fix typo in doc comment

Commit:
Fix user.resetPassword to fail on email not found

Commit:
Merge pull request #1789 from simoami/master

PasswordReset fix for #1374 and #947

Close #1789

sam-github committed:
Refer to licenses with a link

Commit:
lib/registry: fix findModel for model ctor

Fix `registry.findModel(arg)` to support the case when `arg` is already
a model constructor.

bajtos committed:
Merge pull request #1802 from strongloop/fix/findModel

lib/registry: fix findModel for model ctor

Commit:
2.23.0

 * lib/registry: fix findModel for model ctor (Miroslav Bajtoš)
 * Refer to licenses with a link (Sam Roberts)
 * Fix user.resetPassword to fail on email not found (Simo Moujami)
 * Fix typo in doc comment (Rand McKinney)
 * Do not include redundant ports in verify links (Samuel Gaus)
 * Set application's id property only if it's empty. (wusuopu)
 * Check configs for shared method settings (Simon Ho)
 * Add test fixtures for shared methods (Simon Ho)
 * Clean up .jshintrc (Simon Ho)
 * Update comment about user ACL to reflect implementation (Felipe Oliveira Carvalho)

Commit:
Add case-sensitve email option for User model.

Commit:
Fix cookie-parser error

superkhau committed:
Merge pull request #1805 from strongloop/bug/cookie-parser

Fix cookie-parser error

Commit:
2.24.0

 * Fix cookie-parser error (Simon Ho)

Commit:
Fix typo in description of persistedModel.updateAttributes()

richardpringle committed:
Merge pull request #1811 from strongloop/typo-fix

Fix typo in description of persistedModel.updateAttributes()

Commit:
2.25.0

 * Fix typo in description of persistedModel.updateAttributes() (Richard Pringle)

ytang-godaddy committed:
modelName not modelType

shelbys commented:
@y-tang - This has already been fixed on `#upstream/master`, so we should just merge to latest and test that instead.  This fork is mainly to lock to `shelbys/strong-remoting` anyway

y-tang commented:
@shelbys I need this change in to combine shopper internal and external. Without this fix, I am getting errors. Can we merge to latest now?

shelbys committed:
Merge remote-tracking branch 'upstream/master' into Registry-findModel-modelName

* upstream/master: (50 commits)
  2.25.0
  Fix typo in description of persistedModel.updateAttributes()
  2.24.0
  Fix cookie-parser error
  2.23.0
  lib/registry: fix findModel for model ctor
  Refer to licenses with a link
  Fix user.resetPassword to fail on email not found
  Fix typo in doc comment
  Do not include redundant ports in verify links
  Set application's id property only if it's empty.
  Check configs for shared method settings
  Add test fixtures for shared methods
  Clean up .jshintrc
  2.22.2
  Use strongloop conventions for licensing
  Set package license to MIT
  2.22.1
  Fix perf of rectification after updateAttributes
  Update comment about user ACL to reflect implementation
  ...

shelbys committed:
Reverted strong-remoting to avoid problems in loopback-datasource-juggler

Commit:
Update model.js

*Add doc comments for nestRemoting
*change cb to filterCallback in nestRemoting

Commit:
Adding properties description for User Model

- Added description for `realm` `created` `lastUpdated` `status`
- End of line for descriptions to be peroid

bajtos committed:
Change: correctly rectify no-change

Modify `Change.rectify()` to not make any changes to the Change instance
(most notably to not modify the `checkpoint` field) when the tracked
model instance was not changed.

This should improve the performance of change replication as it reduces
the number of unnecessary replications.

For example, before this commit, every run of `rectifyAll` would
trigger a full sync of all clients, because all change instances would
be moved to the current checkpoint.

bajtos committed:
change: skip cp lookup on no change

Modify `Change.rectify` to look up the current checkpoint only when
there was actually some change made.

This should improve the performance of `rectifyAll` when called from a
regular timer and there were no changes made since the last call.
Before this commit, `rectifyAll` would perform N calls of
`Checkpoint.current` where N is the number of model instances. With
this commit in place, no call is made.

Amir-61 committed:
Merge pull request #1847 from strongloop/add-nestRemoting-doc

Update model.js

bajtos committed:
Merge pull request #1855 from strongloop/fix/user-model-docs

Adding properties description for User Model

bajtos committed:
Merge pull request #1804 from richardpringle/master

Add case-sensitive email option for User model

bajtos committed:
Merge pull request #1860 from strongloop/fix/replication-performance

Fix replication performance

bajtos committed:
2.26.0

 * change: skip cp lookup on no change (Miroslav Bajtoš)
 * Change: correctly rectify no-change (Miroslav Bajtoš)
 * Update model.js (Rand McKinney)
 * Adding properties description for User Model (David Cheung)
 * Add case-sensitve email option for User model. (Richard Pringle)

bajtos committed:
PersistedModel: log rectify/rectifyAll triggers

Add debug logging to identify what operations trigger rectify and
rectifyAll calls.

bajtos committed:
Merge pull request #1894 from strongloop/feature/improve-replication-debugging

PersistedModel: log rectify/rectifyAll triggers

bajtos committed:
2.26.1

 * PersistedModel: log rectify/rectifyAll triggers (Miroslav Bajtoš)

Commit:
Fix bulkUpdate to not trigger rectifyAll

Fix `getIdFromWhereByModelId()` to correctly detect the situation
when "bulkUpdate" performs a write operation using a where filter
containing both id attribute but also all other model attributes.

This should significantly improve the performance of change replication,
because the cost of running rectifyAll is very high.

bajtos committed:
Merge pull request #1899 from strongloop/rectifyChange_Vs_rectifyAllChanges

Fix bulkUpdate to not trigger rectifyAll

bajtos committed:
2.26.2

 * Fix bulkUpdate to not trigger rectifyAll (Amir Jafarian)

Commit:
Merge branch '2.x-latest' of https://github.com/strongloop/loopback into Registry-findModel-modelName
